File: typescript/examples/cancel_all.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Cancel All Orders Example
 *
 * Cancel all open orders, or all orders for a specific asset.
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // Check open orders first
  const orders = await sdk.openOrders();
  console.log(`Open orders: ${orders.count}`);

  // Cancel all orders
  const result = await sdk.cancelAll();
  console.log(`Cancel all: ${JSON.stringify(result)}`);

  // Or cancel just BTC orders:
  // await sdk.cancelAll("BTC");
}

main().catch(console.error);

File: python/examples/cancel_all.py
#!/usr/bin/env python3
"""
Cancel All Orders Example

Cancel all open orders, or all orders for a specific asset.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python cancel_all.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Cancel All Orders Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Address: {sdk.address}")

    # Check open orders first
    orders = sdk.open_orders()
    print(f"Open orders: {orders.count}")

    # Cancel all orders
    result = sdk.cancel_all()
    print(f"Cancel all: {result}")

    # Or cancel just BTC orders:
    # sdk.cancel_all("BTC")


if __name__ == "__main__":
    main()
